src/idl_gen_swift.cpp

```diff
@@ -0,0 +1,719 @@
+    if (can_override) {
+      if (type.enum_def)
+        return WrapInNameSpace(type.enum_def->defined_namespace,
+                               Name(*type.enum_def));
+      if (type.base_type == BASE_TYPE_BOOL) return "Bool";
+    }
+    return swift_type[type.base_type];
+  }
+
+  std::string GenEnumDecl(const EnumDef &enum_def) const {
+    return "enum " + Name(enum_def);
+  }
+
+  std::string EscapeKeyword(const std::string &name) const {
+    return keywords_.find(name) == keywords_.end() ? name : name + "_";
+  }
+
+  std::string Name(const EnumVal &ev) const { return EscapeKeyword(ev.name); }
+
+  std::string Name(const Definition &def) const {
+    return EscapeKeyword(MakeCamel(def.name, false));
+  }
+
+  const Namespace *cur_name_space_;
+
+  static std::string GeneratedFileName(const std::string &path,
+                                       const std::string &file_name) {
+    return path + file_name + "_generated.swift";
+  }
+  // MARK: - Copied from the cpp implementation, needs revisiting
+  void SetNameSpace(const Namespace *ns) {
+    if (cur_name_space_ == ns) { return; }
+    if (*should_generate_extension) { return; }
+    // Compute the size of the longest common namespace prefix.
+    // If cur_name_space is A::B::C::D and ns is A::B::E::F::G,
+    // the common prefix is A::B:: and we have old_size = 4, new_size = 5
+    // and common_prefix_size = 2
+    size_t old_size = cur_name_space_ ? cur_name_space_->components.size() : 0;
+    size_t new_size = ns ? ns->components.size() : 0;
+
+    size_t common_prefix_size = 0;
+    while (common_prefix_size < old_size && common_prefix_size < new_size &&
+           ns->components[common_prefix_size] ==
+               cur_name_space_->components[common_prefix_size]) {
+      common_prefix_size++;
+    }
+
+    // Close cur_name_space in reverse order to reach the common prefix.
+    // In the previous example, D then C are closed.
+    for (size_t j = old_size; j > common_prefix_size; --j) {
+      if (namespace_depth != 0) {
+        code_ += "}";
+        namespace_depth -= 1;
+      }
+      mark(cur_name_space_->components[j - 1]);
+    }
+    if (old_size != common_prefix_size) { code_ += ""; }
+
+    // open namespace parts to reach the ns namespace
+    // in the previous example, E, then F, then G are opened
+    bool is_extension = false;
+    for (auto j = common_prefix_size; j != new_size; ++j) {
+      std::string name = ns->components[j];
+      if (namespaces_.find(name) == namespaces_.end()) {
+        code_ += "public enum " + name + " {";
+        namespace_depth += 1;
+        namespaces_.insert(name);
+      } else {
+        code_ += "}";
+        is_extension = true;
+      }
+    }
+    if (*should_generate_extension) {
+      code_.SetValue("EXTENSION", FullNamespace(".", *ns));
+      code_ += "extension {{EXTENSION}} {";
+    }
+    if (new_size != common_prefix_size) { code_ += ""; }
+    should_generate_extension = &is_extension;
```

Comment:
`is_extension` is **local** variable declared at 687.
I would recommend removing the pointer `should_generate_extension` from the code.
I don't understand why this variable is pointer, not value.

Reply:
PVS found it too:
V506 [CWE-562] Pointer to local variable 'is_extension' is stored outside the scope of this variable. Such a pointer will become invalid. idl_gen_swift.cpp 704